src/braket/aws/aws_quantum_task.py

```diff
@@ -257,7 +257,10 @@ def state(self, use_cached_value: bool = False) -> str:
     def _status(self, use_cached_value=False):
         status = self.metadata(use_cached_value).get("status")
         if not use_cached_value and status in self.NO_RESULT_TERMINAL_STATES:
-            self._logger.warning(f"Task is in terminal state {status} and no result is available")
+            self._logger.warning(f"Task is in terminal state {status} and no result is available.")
+            if status == "FAILED":
+                failure_reason = self.metadata(use_cached_value).get("failureReason")
```

Comment:
`.get('failureReason', 'unknown reason')` since sometimes failureReason is not populated

Reply:
The message here has changed recently to support an unpopulated failure reason. Here is the current main error:
```
message = (
                    f"Error retrieving results, your job is in {job_state} state. "
                    "Your job has failed due to: "
                    f"{job_response.get('failureReason', 'unknown reason')}"
                    if job_state == "FAILED"
                    else f"Error retrieving results, your job is in {job_state} state."
                )
```
It's alright to modify this, it's just a template that will make grammatical sense in cases when the reason is not given

Reply:
It looks like the latest main changes haven't been pulled in here. That might be helpful